controllers/numaresourcesoperator_controller_test.go

```diff
@@ -138,224 +140,237 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 
-						mc := &machineconfigv1.MachineConfig{}
-						key = client.ObjectKey{
-							Name: rte.GetMachineConfigName(nro.Name, mcpWithComplexMachineConfigSelector.Name),
-						}
-						Expect(reconciler.Client.Get(context.TODO(), key, mc)).ToNot(HaveOccurred())
+						Expect(reconciler.Client.Get(context.TODO(), nroKey, nro)).ToNot(HaveOccurred())
+						Expect(len(nro.Status.MachineConfigPools)).To(Equal(1))
+						Expect(nro.Status.MachineConfigPools[0].Name).To(Equal("test1"))
 					})
 				})
 
-				When("machine config selector does not match machine config labels", func() {
+				When("machine config pools are ready", func() {
 					BeforeEach(func() {
-						mcpWithComplexMachineConfigSelector.Spec.MachineConfigSelector = &metav1.LabelSelector{
-							MatchExpressions: []metav1.LabelSelectorRequirement{
-								{
-									Key:      rte.MachineConfigLabelKey,
-									Operator: metav1.LabelSelectorOpIn,
-									Values:   []string{"worker", "worker-cnf"},
-								},
+						var err error
+
+						// Ensure mcp1 is ready
+						Expect(reconciler.Client.Get(context.TODO(), client.ObjectKeyFromObject(mcp1), mcp1)).ToNot(HaveOccurred())
+						mcp1.Status.Configuration.Source = []corev1.ObjectReference{
+							{
+								Name: rte.GetMachineConfigName(nro.Name, mcp1.Name),
 							},
 						}
-					})
+						mcp1.Status.Conditions = []machineconfigv1.MachineConfigPoolCondition{
+							{
+								Type:   machineconfigv1.MachineConfigPoolUpdated,
+								Status: corev1.ConditionTrue,
+							},
+						}
+						Expect(reconciler.Client.Status().Update(context.TODO(), mcp1))
 
-					It("should not create the machine config and set the degraded condition", func() {
-						reconciler, err := NewFakeNUMAResourcesOperatorReconciler(platform.OpenShift, nro, mcpWithComplexMachineConfigSelector)
+						// ensure mcp2 is ready
+						Expect(reconciler.Client.Get(context.TODO(), client.ObjectKeyFromObject(mcp2), mcp2)).ToNot(HaveOccurred())
+						mcp2.Status.Configuration.Source = []corev1.ObjectReference{
+							{
+								Name: rte.GetMachineConfigName(nro.Name, mcp2.Name),
+							},
+						}
+						mcp2.Status.Conditions = []machineconfigv1.MachineConfigPoolCondition{
+							{
+								Type:   machineconfigv1.MachineConfigPoolUpdated,
+								Status: corev1.ConditionTrue,
+							},
+						}
+						Expect(reconciler.Client.Status().Update(context.TODO(), mcp2))
+
+						secondLoopResult, err = reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: nroKey})
 						Expect(err).ToNot(HaveOccurred())
+					})
+					It("should continue with creation of additional components", func() {
+						// check reconcile second loop result
+						Expect(secondLoopResult).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+
+						// Check All the additional components are created
+						key := client.ObjectKey{
+							Name:      "rte",
+							Namespace: testNamespace,
+						}
```

Comment:
I'd NOT reuse objects, it's bad practice, even if all the names are the same. In this case I prefer to create temporary object keys for each object (there is a helper already for this, see ObjectKeyFromObject)

Reply:
I also prefer not to reuse them ... but was just trying not to modify to much of the existing code in the reorganize process...

Reply:
it's fine, as long as you do refactoring on it's own commit. Better that than reusing objects :)

Reply:
I have already changed some names in the previous commit so in the end I have changed them all ...